rust/nix-expr/src/eval_state.rs

```diff
@@ -131,6 +137,46 @@ impl EvalState {
         }
         self.get_string(value)
     }
+    pub fn realise_string(
+        &self,
+        value: &Value,
+        is_import_from_derivation: bool,
+    ) -> Result<RealisedString> {
+        let t = self.value_type(value)?;
+        if t != ValueType::String {
+            bail!("expected a string, but got a {:?}", t);
+        }
+
+        let rs = unsafe {
```

Comment:
The docs say that the realised string returned from `string_realise()` need to be freed with `realised_string_free()`. I don't see that we do that anywhere, However the storage gets reused to make the `String` we return. So the memory will get freed, but not in the way specified by the API.

This seems like either a problem now or a potential problem in the future, but if this is the intention it should at least be documented that's what we're doing.

The obvious thing to do is `impl Drop for RealisedString` and then deconstruct the `String` there and call `realised_string_free()`. But it's more complicated than that because someone could take the `String` out of `RealisedString` unless we make it private.

Reply:
I had decided to eagerly convert it to Rust struct for simplicity, but forgot to free the `nix_realised_string`.
I don't think we need give lazy access, because in Nix itself the return type is not lazy, so by converting eagerly, we don't have to hold on to the `nix_realised_string`. (Also realisations are quite heavy, so no need to micro-optimize the conversions either)